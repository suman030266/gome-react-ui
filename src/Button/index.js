import React,{Component} from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import './index.scss';

export default class Button extends Component{
  constructor(props){
    super();
    this.state = {
      type: props.type || 'default'
    };
  }
  /*static defaultProps = {
    state: 'able'
  }
  static propTypes = {
      type: PropTypes.string,
      state: PropTypes.oneOf(['disable', 'able']),
      shape: PropTypes.oneOf(['circle', 'circle-outline']),
      size: PropTypes.oneOf(['large', 'default', 'small']),
      onClick: PropTypes.func,
      className: PropTypes.string,
  }*/
  _buttonClicked(e) {
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
  render(){
    let obj = classnames({
      "gome-btn": true,
      "gome-btn-def": this.state.type == 'default',
      "gome-btn-primary": this.props.type == 'primary',
      "gome-btn-dashed": this.props.type == 'dashed',
      "gome-btn-danger": this.props.type == 'danger',
      "gome-btn-large": this.props.size == 'large',
      "gome-btn-small": this.props.size == 'small',
      "gome-btn-disable": this.props.state == 'disable'
    });
    return (
      <button className={obj} onClick={this._buttonClicked.bind(this)}>
        <span>{this.props.children || '按钮'}</span>
      </button>
    )
  }
}
