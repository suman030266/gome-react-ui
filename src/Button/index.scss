.gome-btn {
  outline: none;
  height: 32px;
  line-height: 32px;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  -ms-touch-action: manipulation;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid transparent;
  white-space: nowrap;
  padding: 0 15px;
  font-size: 14px;
  border-radius: 4px;
  height: 32px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
  transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
  position: relative;
  color: rgba(0, 0, 0, 0.65);
  background-color: #fff;
  border-color: #d9d9d9;
  font-family: -apple-system,"Helvetica Neue",Helvetica,Arial,"PingFang SC","Hiragino Sans GB","WenQuanYi Micro Hei","Microsoft Yahei",sans-serif;

  &.gome-btn-primary {
    color: #fff;
    background-color: #1890ff;
    border-color: #1890ff;
  }
  &.gome-btn-primary:hover{
    background-color: #40a9ff;
    border-color: #40a9ff;
  }
  &.gome-btn-primary:active{
    background-color: #096dd9;
    border-color: #096dd9;
    transition: none;
  }
  &.gome-btn-clicked:after {
    content: '';
    position: absolute;
    top: -1px;
    left: -1px;
    bottom: -1px;
    right: -1px;
    border-radius: inherit;
    border: 0 solid #1890ff;
    opacity: 0.4;
    -webkit-animation: buttonEffect .4s;
            animation: buttonEffect .4s;
    display: block;
  }
  &.gome-btn-danger.gome-btn-clicked:after {
    border-color: #f5222d;
  }
  &.gome-btn-dashed {
    color: rgba(0, 0, 0, 0.65);
    background-color: #fff;
    border-color: #d9d9d9;
    border-style: dashed;
  }
  &.gome-btn-danger {
    color: #f5222d;
    background-color: #f5f5f5;
    border-color: #d9d9d9;
  }
  &.gome-btn-large {
    height: 40px;
    line-height: 40px;
    padding: 0 20px;
  }
  &.gome-btn-small {
    height: 20px;
    line-height: 20px;
    padding: 0 10px;
  }
  &.gome-btn-disable,
  &.gome-btn-disable:active,
  &.gome-btn-disable.active,
  &.gome-btn-disable:hover {
    color: rgba(0, 0, 0, 0.25);
    background-color: #f5f5f5;
    border-color: #d9d9d9;
    cursor: not-allowed;
  }
}
