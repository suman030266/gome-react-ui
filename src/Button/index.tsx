import * as React from "react";
import * as classNames from 'classnames';
import * as PropTypes from 'prop-types';
import './index.scss';
// enum State{ 'disable', 'able' };
// enum Shape{ 'circle', 'circle-outline' };
// enum Size{ 'large', 'default', 'small' };

interface Props {
  type?: string;
  states?: string;
  shape?: string;
  size?: string;
  onClick: any;
  className?: string;
  // children: any;
}

interface State  {
  type?: string;
  states?: string;
  shape?: string;
  size?: string;
  // children: any;
  className?: string;
  onClick?: any;
  "gome-btn-def"?: boolean;
}

export default class Button extends React.Component<Props, State> {
  public state : State;
  constructor(props: Props){
    super(props);
    this.state = {
      type: props.type || 'default',
      shape: props.shape || '',
      size: props.size || '',
      "gome-btn-def": props.type === 'default',
      states: "able",
      className: props.className || '',
      onClick: props.onClick || ''
      // children: props.children
    };
  }
  /*static defaultProps = {
    state: 'able'
  }
  */
  public _buttonClicked(e) {
    if (this.props.onClick) {
      this.props.onClick(e);
    }
  }
  public render(){
    let { type, size, states } = this.state;
    let { children } = this.props;
    let obj = classNames({
      "gome-btn": true,
      "gome-btn-def": type == 'default',
      "gome-btn-primary": type == 'primary',
      "gome-btn-dashed": type == 'dashed',
      "gome-btn-danger": type == 'danger',
      "gome-btn-large": size == 'large',
      "gome-btn-small": size == 'small',
      "gome-btn-disable": states == 'disable'
    });
    return (
      <button className={obj} onClick={this._buttonClicked.bind(this)}>
        <span>{children || '按钮'}</span>
      </button>
    )
  }
}
