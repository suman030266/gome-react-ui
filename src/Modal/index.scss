*{
  padding: 0;
  margin: 0;
}
body{
  font-family: 'Microsoft Yahei', '微软雅黑', 'PingFang SC', 'Hiragino Sans GB', '黑体', sans-serif;
}
.hide{
  display: none;
}
.gome-modal{
  height: 100%;
  width: 100%;
  position: fixed;
  left: 0;
  top: 0;
  .gome-modal-mask{
    background-color: rgba(0, 0, 0, .65);
    height: 100%;
    width: 100%;
    position: fixed;
  }
  .gome-modal-content{
    /*z-index: 2;
    position: fixed;
    background-color: #fff;
    left: 0;
    top: 0;
    bottom: 0;
    right: 0;
    margin: auto;
    line-height: 28px;
    border-radius: 4px;
    box-shadow: 0 4px 12px rgba(0, 0, 0, 0.15);
    overflow: hidden;
    */
    max-height: 100%;
    min-height: 100px;
    min-width: 300px;
    max-width: 100%;
    position: fixed;
    top: 50%;
    left: 50%;
    border-radius: 4px;
    background-color: #fff;
    -webkit-transform: translate(-50%,-50%);
    -moz-transform: translate(-50%,-50%);
    -o-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
    .gome-modal-close{
      position: absolute;
      right: 15px;
      top: 15px;
      height: 20px;
      width: 20px;
      line-height: 20px;
      font-size: 20px;
      color: #888;
      text-align: center;
      cursor: pointer;
    }
    h3.gome-modal-header{
      padding: 16px 24px;
      border-radius: 4px 4px 0 0;
      background: #fff;
      color: rgba(0, 0, 0, 0.65);
      span{
        display: inline-block;
        margin: 0;
        font-size: 16px;
        line-height: 22px;
        font-weight: 500;
        color: rgba(0, 0, 0, 0.85);
      }
    }
    .gome-modal-body{
      border-top: 1px solid #eee;
      border-bottom: 1px solid #eee;
      padding: 24px;
      font-size: 14px;
      line-height: 1.5;
    }
    .gome-modal-foot{
      padding: 10px 16px;
      div{
        text-align: right;
        button{
          margin-left: 8px;
        }
      }
    }
  }
}
