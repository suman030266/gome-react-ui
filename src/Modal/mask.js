import React,{Component} from 'react';

export default class Mask extends React.Component{
  constructor(props){
    super();
  }
  render(){
    let {clickCb, canClick} = this.props;
    return (
      <div onClick={canClick ? clickCb.bind(this) : null}  className="gome-modal-mask"></div>
    )
  }
}
