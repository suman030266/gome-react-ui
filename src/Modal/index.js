import React,{Component} from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import Button from '../Button';
import './index.scss';
import Mask from './mask';

export default class Model extends React.Component{
  constructor(props){
    super();
    this.state = {
      visible: props.visible,  // 是否展示 默认不展示
      width: props.width != undefined ? (props.width + 'px') : '520px',  // 弹层宽度 默认520px
      height: props.height != undefined ? (props.height + 'px') : '350px',  // 弹层高度 默认520px
      closeBtn: props.closeBtn != undefined ? props.closeBtn : true, // 关闭按钮 默认展示
      title: props.title || '提示',  // 弹窗title 默认 提示
      footerType: props.footerType || 'default', // 弹窗底部 none/self/default 默认 展示默认底部
      footer: props.footer || [], // 弹窗底部 默认空
      children: props.children,  // 弹窗内容 必传
      canMaskClick: props.canMaskClick != undefined ? props.canMaskClick : true, // 弹窗遮罩是否可点击 默认可点击
      autoClose: props.autoClose != undefined ? props.autoClose : false, // 弹窗是否自动关闭 默认不自动关闭
      closeTimeout: props.closeTimeout || 3000 // 自动关闭时间 默认3000毫秒
    };
  }
  componentDidMount(){
    if(this.state.autoClose){
      setTimeout(()=>{
        this.setState({
          visible: false
        });
      }, this.state.closeTimeout);
    }
  }
  componentWillReceiveProps(newProps, newState){
    this.setState({
      visible: newProps.visible
    });
  }
  closeModal(cb){
    this.setState({
      visible: false
    },()=>{
      cb && cb();
    });
  }
  handleClose(){
    this.closeModal();
  }
  handleCancel(){
    console.log('默认 cancel');
    this.props.onCancel();
    // this.closeModal();
  }
  handleOk(){
    console.log('默认 ok');
    this.props.onOk();
    // this.closeModal(this.props.onOk);
  }
  render(){
    let { closeBtn, title, footerType, footer, children, height, width, canMaskClick, visible } = this.state;
    return (
      <div className={classnames({
        "gome-modal": true,
        "hide": !visible
      })}>
        <Mask clickCb={this.handleClose.bind(this)} canClick={canMaskClick} />
        <div id="modalContent" className="gome-modal-content" style={{width, height}}>
          {closeBtn ? (<a className="gome-modal-close" onClick={this.handleClose.bind(this)}>x</a>) : null}
          <h3 className="gome-modal-header">
            <span>{title}</span>
          </h3>
          <div className="gome-modal-body">
            {children ? (Object.prototype.toString.call(children) === '[object Array]' ? children.map((item)=>item) : children) : null}
          </div>
          {footerType !== 'none' ? (
            <div className="gome-modal-foot">
              {footerType !== 'self' ? (
                <div>
                  <Button className="ok" onClick={this.handleOk.bind(this)}>ok</Button>
                  {/*<Button className="cancel" onClick={this.handleCancel.bind(this)} type="primary">Cancel</Button>*/}
                </div>
              ) : (footer.length && <div>{footer.map((item)=>item)}</div>)
              }
            </div>) : null
          }
        </div>
      </div>
    );
  }
}
