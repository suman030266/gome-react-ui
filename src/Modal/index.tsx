import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as classNames from 'classnames';
import './index.scss';
import Button from '../Button';

import Mask from './mask';

interface Props {
  id: number;
  visible : boolean;
  width?: string;
  height?: any;
  closeBtn?: boolean;
  title?: string;
  footerShow?: boolean;
  children: any;
  canMaskClick?: boolean;
  autoClose?: boolean;
  closeTimeout?: number;
  onCancel: any;
  onOk: any;
  autoCloseCb?: any;
  btnType?: string;
  okText?: string;
  cancelText?: string;
  btnArr?: any;
  dong?: boolean;
}

interface State {
  id: number;
  visible : boolean;
  width?: string;
  height?: any;
  closeBtn?: boolean;
  title?: string;
  footerShow?: boolean;
  children: any;
  canMaskClick?: boolean;
  autoClose?: boolean;
  closeTimeout?: number;
  autoCloseCb?: any;
  timer: any;
  btnType?: string;
  okText?: string;
  cancelText?: string;
  btnArr?: any;
  dong?: boolean;
}

interface Utils {
  handleClose: any;
  handleCancel: any;
  handleOk: any;
}

export default class Modal extends React.Component<Props, State> {
  public state : State;
  public wrap : any;
  public ele : any;
  // public utils : Utils;
  constructor(props: Props){
    super(props);
    this.state = {
      id: props.id,  // 每个弹窗都有自己的id 是一个传入的随机数
      visible: props.visible,  // 是否展示 默认不展示
      width: props.width != undefined ? (props.width + 'px') : null,  // 弹层宽度
      height: props.height != undefined ? (props.height + 'px') : null,  // 弹层高度
      closeBtn: props.closeBtn != undefined ? props.closeBtn : true, // 关闭按钮 默认展示
      title: props.title || '提示',  // 弹窗title 默认 提示
      footerShow: props.footerShow || true, // 弹窗底部 true/false 是否展示 默认true
      children: props.children,  // 弹窗内容 必传
      canMaskClick: props.canMaskClick != undefined ? props.canMaskClick : true, // 弹窗遮罩是否可点击 默认可点击
      autoClose: props.autoClose != undefined ? props.autoClose : false, // 弹窗是否自动关闭 默认不自动关闭
      closeTimeout: props.closeTimeout || 3000, // 自动关闭时间 默认3000毫秒
      autoCloseCb: props.autoCloseCb, // 自动关弹层后回调
      timer: null, // 存储自动关闭弹窗定时器
      btnType: props.btnType || 'sure',  // 按钮展示类型 默认只展示一个确定按钮
      okText: props.okText || '确定',  // 确定按钮文字 默认确定
      cancelText: props.cancelText || '取消', // 取消按钮文字 默认取消
      btnArr: props.btnArr || [],  // 自定义按钮可传入一个数组
    };
    this.wrap = null;
    // this.utils;
    // this.hadlesArr: any[] = ['Close', 'Cancel', 'Ok'];
  }
  public append(){
    if(this.state.visible){
      this.wrap = document.createElement('div');
      this.wrap.className = 'ele' + this.props.id;
      ReactDOM.render(this._render(), this.wrap);
      document.body.appendChild(this.wrap);
      if(document.getElementsByClassName('ele' + this.state.id).length == 2){
        document.body.removeChild(document.getElementsByClassName('ele' + this.state.id)[1]);
      }
      if(this.state.autoClose){
        this.state.timer = setTimeout(()=>{
          this.closeModal();
          this.state.autoCloseCb && this.state.autoCloseCb();
        }, this.state.closeTimeout);
      }
    }else{
      console.log('append 不让元素创建');
    }
  }
  public componentDidMount(){
    console.log('componentDidMount');
    this.append();
    // this.hadlesArr.forEach(item =>{
    //   this.utils[`handle ${item}`] = this.handle(item);
    // });
  }
  public clearTimeouts(){
    clearTimeout(this.state.timer);
  }
  public componentWillUnmount(){
    this.clearTimeouts();
  }
  public componentWillReceiveProps(newProps, newState){
    let eles = document.getElementsByClassName('ele' + newProps.id)[0];
    console.log(eles, newProps.visible);
    console.dir(newProps.id);
    if(newProps.visible){
      this.setState({
        visible: newProps.visible,
        id: newProps.id
      }, ()=>{
        console.log('接收到新的props 要创建组件了');
        this.append();
      });
    }else{
      console.log('销毁');
      this.setState({
        id: newProps.id
      }, ()=>{
        this.closeModal();
      });

    }
  }
  public closeModal(cb=()=>{}){
    try{
      document.body.removeChild(document.getElementsByClassName('ele' + this.state.id)[0]);
    }catch(e){
      console.log(e);
    }
    cb && cb();
  }
  public handleClose(){
    this.closeModal();
    this.clearTimeouts();
  }
  public handleCancel(){
    console.log('默认 cancel回调');
    this.props.onCancel && this.props.onCancel();
    this.handleClose();
  }
  public handleOk(){
    console.log('默认 ok回调');
    this.props.onOk && this.props.onOk();
    this.handleClose();
  }
  // 高阶函数 待修改
  // public handle(str){
  //   return ()=>{
  //     console.log('默认 ${str}回调');
  //     this.props[`on${str}`] && this.props[`on${str}`]();
  //     this.clearTimeouts();
  //   }
  // }
  public _render(){
    let { closeBtn, title, footerShow, height, width, canMaskClick, visible, btnType, okText, cancelText, btnArr } = this.state;
    let { children } = this.props;
    return (
      <div className={classNames('gome-modal', { "hide": !visible })} >
        <Mask clickCb={this.handleClose.bind(this)} canClick={canMaskClick} />
        <div id="modalContent" className="gome-modal-content" style={{width, height}}>
          {closeBtn ? (<a className="gome-modal-close" onClick={this.handleClose.bind(this)}>x</a>) : null}
          <h3 className="gome-modal-header">
            <span>{title}</span>
          </h3>
          <div className="gome-modal-body">
            {children ? (Object.prototype.toString.call(children) === '[object Array]' ? children.map((item)=>item) : children) : null}
          </div>
          {footerShow ? (
            <div className="gome-modal-foot">
              {
                (btnType == 'sure' || btnType == 'cancel' || btnType == 'both') && !btnArr.length ? (
                  btnType == 'sure' ? <Button className="ok" onClick={this.handleOk.bind(this)}>{okText}</Button> : (
                    btnType == 'cancel' ? (
                      <Button className="cancel" onClick={this.handleCancel.bind(this)} type="primary">{cancelText}</Button>
                    ) : btnType == 'both' ? (
                      <React.Fragment>
                        <Button className="ok" onClick={this.handleOk.bind(this)}>{okText}</Button>
                        <Button className="cancel" onClick={this.handleCancel.bind(this)} type="primary">{cancelText}</Button>
                      </React.Fragment>
                    ) : null
                  )
                ) : (
                  btnArr.length && (btnArr.map( item =>{
                    return <Button key={item.index} {...item} className={item.classNames} onClick={item.click.bind(this)}>{item.text}</Button>;
                  }))
                )
              }
              {/*footerShow !== 'self' ? (
                <div>
                  {<Button className="ok" onClick={this.handleOk.bind(this)}>确定</Button>}
                  {<Button className="cancel" onClick={this.handleCancel.bind(this)} type="primary">取消</Button>}
                </div>
              ) : (footer.length && <div>{footer.map((item)=>item)}</div>)
              */}
            </div>) : null
          }
        </div>
      </div>
    );
  }
  public render(){
    return null;
  }
}
