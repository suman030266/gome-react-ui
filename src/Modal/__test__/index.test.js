import React from 'react';
import ReactDOM from 'react-dom';
import Modal from '../index';
import Button from '../../Button/index';
import { shallow, mount, render, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-dom/test-utils';

configure({ adapter: new Adapter() });

// 测试组件是否正常挂载
(function (){
  const wrapper = shallow(
    <Modal
      visible={true}
      footerType={'none'}
    >
      <p>第1行文字</p>
      <p>第2行文字</p>
    </ Modal>);
  describe('测试组件是否正常挂载', () => {
    it('Modal Component should be render', () => {
      expect(wrapper.find('.gome-modal-content').exists());
      expect(!wrapper.find('.gome-modal-foot').exists());
    })
  });
})();

test('测试第1个 自定义标题 head不传入 无尾部 无宽度', () => {
  const component = renderer.create(
    <Modal
      visible={true}
      title={"第一个弹窗"}
      footerType={'none'}
    >
      <p>第1行文字</p>
      <p>第2行文字</p>
    </ Modal>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // console.log('tree----before');
  // console.log(JSON.stringify(tree));
  // console.log('tree----after');
});

test('测试第2个 默认标题 尾部展示默认 宽600', () => {
  const component = renderer.create(
    <Modal
      visible={true}
      footerType={'default'}
      width={600}
    >
      <p>第1行文字</p>
      <p>第2行文字</p>
      <p>第3行文字</p>
      <p>第4行文字</p>
    </ Modal>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // console.log('tree----before');
  // console.log(JSON.stringify(tree));
  // console.log('tree----after');
});

test('测试第3个自定义展示尾部', () => {
  const component = renderer.create(
    <Modal
      visible={true}
      footerType={'self'}
      width={800}
      footer={[
        <Button key="back">return</Button>,
        <Button key="submit" type="primary">submit</Button>
      ]}
    >
      <p>111</p>
    </ Modal>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // console.log('tree----before');
  // console.log(JSON.stringify(tree));
  // console.log('tree----after');
});

// 点击关闭
(function (){
  const wrapper = shallow(
    <Modal
      visible={true}
      title={"即将关闭"}
      footerType={'none'}
    >
      <p>第1行文字</p>
    </ Modal>
  );
  expect(wrapper.find('.gome-modal.hide').length).toEqual(0);
  wrapper.find('.gome-modal-close').simulate('click');
  expect(wrapper.find('.gome-modal.hide').length).toEqual(1);
})();




test('测试第3个自定义展示尾部', () => {
  const component = renderer.create(
    <Modal
      visible={true}
      footerType={'default'}
      footer={[
        <Button key="back">return</Button>,
        <Button key="submit" type="primary">submit</Button>
      ]}
      title={"即将关闭"}
      onOk={()=>{console.log('ooooooo');}}
      onCancel={()=>{console.log('ccccccc');}}
    >
      <p>第1行文字</p>
    </ Modal>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  // console.log('tree----before');
  console.log(JSON.stringify(tree));
  // console.log('tree----after');
});

// 点击确定
(function (){
  const wrapper = mount(
    <Modal
      visible={true}
      footerType={'default'}
      footer={[
        <Button key="back">return</Button>,
        <Button key="submit" type="primary">submit</Button>
      ]}
      title={"即将关闭"}
      onOk={()=>{console.log('ooooooo');}}
      onCancel={()=>{console.log('ccccccc');}}
    >
      <p>第1行文字</p>
    </ Modal>
  );
  expect(wrapper.find('.gome-btn.gome-btn-def').length).toEqual(1);
  wrapper.find('.gome-btn.gome-btn-def').simulate('click');
  expect(wrapper.find('.gome-btn.gome-btn-primary').length).toEqual(1);
  expect(wrapper.find('.gome-modal.hide').length).toEqual(1);
})();

// 点击取消
(function (){
  const wrapper = mount(
    <Modal
      visible={true}
      footerType={'default'}
      footer={[
        <Button key="back">return</Button>,
        <Button key="submit" type="primary">submit</Button>
      ]}
      title={"即将关闭"}
      onOk={()=>{console.log('ooooooo');}}
      onCancel={()=>{console.log('ccccccc');}}
    >
      <p>第1行文字</p>
    </ Modal>
  );
  expect(wrapper.find('.gome-btn.gome-btn-def').length).toEqual(1);
  expect(wrapper.find('.gome-btn.gome-btn-primary').length).toEqual(1);
  wrapper.find('.gome-btn.gome-btn-primary').simulate('click');
  expect(wrapper.find('.gome-modal.hide').length).toEqual(1);
})();


























//
