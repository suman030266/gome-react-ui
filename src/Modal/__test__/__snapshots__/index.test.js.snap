// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试第1个 自定义标题 head不传入 无尾部 无宽度 1`] = `
<div
  className="gome-modal"
>
  <div
    className="gome-modal-mask"
  />
  <div
    className="gome-modal-content"
    id="modalContent"
    style={
      Object {
        "width": "520px",
      }
    }
  >
    <a
      className="gome-modal-close"
      onClick={[Function]}
    >
      x
    </a>
    <h3
      className="gome-modal-header"
    >
      <span>
        第一个弹窗
      </span>
    </h3>
    <div
      className="gome-modal-body"
    >
      <p>
        第1行文字
      </p>
      <p>
        第2行文字
      </p>
    </div>
  </div>
</div>
`;

exports[`测试第2个 默认标题 尾部展示默认 宽600 1`] = `
<div
  className="gome-modal"
>
  <div
    className="gome-modal-mask"
  />
  <div
    className="gome-modal-content"
    id="modalContent"
    style={
      Object {
        "width": "600px",
      }
    }
  >
    <a
      className="gome-modal-close"
      onClick={[Function]}
    >
      x
    </a>
    <h3
      className="gome-modal-header"
    >
      <span>
        默认title
      </span>
    </h3>
    <div
      className="gome-modal-body"
    >
      <p>
        第1行文字
      </p>
      <p>
        第2行文字
      </p>
      <p>
        第3行文字
      </p>
      <p>
        第4行文字
      </p>
    </div>
    <div
      className="gome-modal-foot"
    >
      <div>
        <button
          className="gome-btn gome-btn-def"
          onClick={[Function]}
        >
          <span>
            Cancel
          </span>
        </button>
        <button
          className="gome-btn gome-btn-primary"
          onClick={[Function]}
        >
          <span>
            ok
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`测试第3个自定义展示尾部 1`] = `
<div
  className="gome-modal"
>
  <div
    className="gome-modal-mask"
  />
  <div
    className="gome-modal-content"
    id="modalContent"
    style={
      Object {
        "width": "800px",
      }
    }
  >
    <a
      className="gome-modal-close"
      onClick={[Function]}
    >
      x
    </a>
    <h3
      className="gome-modal-header"
    >
      <span>
        默认title
      </span>
    </h3>
    <div
      className="gome-modal-body"
    >
      <p>
        111
      </p>
    </div>
    <div
      className="gome-modal-foot"
    >
      <div>
        <button
          className="gome-btn gome-btn-def"
          onClick={[Function]}
        >
          <span>
            return
          </span>
        </button>
        <button
          className="gome-btn gome-btn-primary"
          onClick={[Function]}
        >
          <span>
            submit
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`测试第3个自定义展示尾部 2`] = `
<div
  className="gome-modal"
>
  <div
    className="gome-modal-mask"
  />
  <div
    className="gome-modal-content"
    id="modalContent"
    style={
      Object {
        "width": "520px",
      }
    }
  >
    <a
      className="gome-modal-close"
      onClick={[Function]}
    >
      x
    </a>
    <h3
      className="gome-modal-header"
    >
      <span>
        即将关闭
      </span>
    </h3>
    <div
      className="gome-modal-body"
    >
      <p>
        第1行文字
      </p>
    </div>
    <div
      className="gome-modal-foot"
    >
      <div>
        <button
          className="gome-btn gome-btn-def"
          onClick={[Function]}
        >
          <span>
            Cancel
          </span>
        </button>
        <button
          className="gome-btn gome-btn-primary"
          onClick={[Function]}
        >
          <span>
            ok
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
